scripts/scripts.js

```diff
@@ -200,6 +200,37 @@ export function formatDate(date) {
   });
 }
 
+function addBlogTopics(main) {
+  const sidebarBolgTopic = main.querySelector('.blog-topics');
+  const apiUrl = '/blogs/query-index.json?sheet=topics';
+  if (sidebarBolgTopic) {
+    fetch(apiUrl)
+      .then((response) => {
```

Comment:
We use the `async/await` syntax in AEM EDS code as a best practice, rather than `.then`. This allows to have more find grained control when to let the code advance to the next operations.

so in this case `const reponse = await fecth(...)` and same below when retrieving `response.json()`.

Reply:
You can see an example here: https://github.com/hlxsites/servicenow/blob/main/scripts/scripts.js#L35-L48
(or you can even call that function directly). 

Alternatively there is also a very small lib: https://github.com/hlxsites/servicenow/blob/main/scripts/ffetch.js that can take care of this for you and you would do:
```javascript
// retrieve data
const blogTopics = await ffetch('/query-index.json')
  .sheet('products')
  .filter(...) //exclude other locales
  .all();
  
// render
block.append(...(blogTopics).map(...));
```